src/scripts/pagerduty.coffee

```
@@ -864,20 +938,32 @@ module.exports = (robot) ->
     query = {
       since: now,
       until: oneHour,
-      overflow: 'true'
+      schedule_ids: [schedule.id]
     }
-    pagerduty.get "/schedules/#{schedule.id}/entries", query, (err, json) ->
+    pagerduty.get "/oncalls", query, (err, json) ->
       if err?
         cb(err, null, null)
         return
 
-      unless json.entries and json.entries.length > 0
+      unless json.oncalls and json.oncalls.length > 0
         cb(null, "nobody", schedule)
         return
 
-      cb(null, json.entries[0].user, schedule)
+      userId = json.oncalls[0].user.id
+      getPagerDutyUser userId, (user) ->
+        cb(null, user, schedule)
+
+  getPagerDutyUser = (userId, cb) ->
+    pagerduty.get "/users/#{userId}", (err, json) ->
+      if err?
+        cb(null)
```

There needs to be a `return` after this line so it does’t fall through 😄 

Do you think this function should bubble the error up by invoking `cb` with two args? I tend to err on that side to allow higher level error handling 🤔  I think this may even be needed in the `withCurrentOncallUser` case above.